### PR TITLE
Move Away from Docker Based action to NodeJS based action

### PR DESCRIPTION
**Pull request by justinbarclay**

### CODE REVIEW

**Comment by justinbarclay** [comment]
@SamDesmondKing this is ready for review.
I think the problem with code-analyzer results is that we don't (or can't) support v3 package-lock.json files in a github action and this repo is written using a v3 package-lock file

File: LICENSE
Comment (SamDesmondKing) [inline LICENSE:1]: We probably don't need this, I assume it's copied over from the typescript action template

File: README.md [hunk 1]
```diff
@@ -37,21 +46,25 @@ jobs:
...
-    - uses: tidalmigrations/tidal-code-action@v0
-      with:
-        tidal-email: ${{ secrets.TIDAL_EMAIL }}
-        tidal-password: ${{ secrets.TIDAL_PASSWORD }}
-        tidal-url: https://workspace.tidal.cloud
-        app-id: 1
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - uses: tidalmigrations/tidal-code-action@v2
```
Comment (SamDesmondKing) [inline README.md:52]: We skipped v1?

File: __tests__/main.test.ts
Comment (SamDesmondKing) [inline __tests__/main.test.ts:1]: Nice tests

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+inputs:
+  tidal-email:
+    description: "The email for the Tidal Accelerator service user"
+    required: true
+  tidal-password:
+    description: "The password for the Tidal Accelerator service user"
+    required: true
+  tidal-url:
+    description:
+      "The workspace for url for your Tidaol Accelerator Account ex:
```
Comment (SamDesmondKing) [inline action.yml:13]: Typo here

File: script/release
Comment (SamDesmondKing) [inline script/release:1]: This is nice to have, I think I did this manually here at some point

File: src/main.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+export const analyzeCode = async () => {
+  const tidalEmail = getInput("tidal-email");
+  const tidalPassword = getInput("tidal-password");
+  const tidalUrl = getInput("tidal-url");
+  const directory = toPlatformPath(getInput("directory"));
+  const appId = getInput("app-id");
+  const pathToCli = await setup();
+  debug("Running `tidal analyze code` command.`");
+
+  await exec(`${pathToCli}/tidal`, [
```
Comment (SamDesmondKing) [inline src/main.ts:116]: Man, I didn't realize node could even run cli commands

**Review by SamDesmondKing — COMMENTED** [review]
Looks great! Only thing that might need a fixup is a typo in a description, IFF that's exposed to the user.

**Comment by SamDesmondKing** [comment]
Also as a thought, since this is public-facing, we may want to use github's release UI rather than tags, that way it's clear for the user when they visit the repo what the latest version is. Right now the latest version is `1.1`.

image

File: README.md (hunk 1 above)
Comment (justinbarclay) [reply README.md:52]: Nah, just our docs were out of date.
https://github.com/tidalmigrations/tidal-code-action/releases/tag/v1 😅

File: src/main.ts (hunk 3 above)
Comment (justinbarclay) [reply src/main.ts:116]: Every programming language should be able to :)
All they have to do is reach out to exec
https://en.wikipedia.org/wiki/Exec_(system_call)

File: action.yml (hunk 2 above)
Comment (justinbarclay) [reply action.yml:13]: good catch

File: script/release
Comment (justinbarclay) [reply script/release:1]: I still tag manually, which is still easier than running a script. I am not sure the script is long enough for this world, though. People who like using the command line will just use `git tag` directly, and those who don't will just click the tag/release button.

Comment (justinbarclay) [reply script/release:1]: Did you have trouble with releasing last time because the process wasn't documented?

**Pull request merged**